contrib/storage-http/src/main/java/org/apache/drill/exec/store/http/HttpJsonOptions.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.drill.exec.store.http;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import lombok.Builder;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.ToString;
+import lombok.experimental.Accessors;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.drill.exec.ExecConstants;
+import org.apache.drill.exec.server.options.OptionSet;
+import org.apache.drill.exec.store.easy.json.loader.JsonLoaderOptions;
+
+@Slf4j
+@Builder
+@Getter
+@Accessors(fluent = true)
+@EqualsAndHashCode
+@ToString
+@JsonInclude(JsonInclude.Include.NON_DEFAULT)
+@JsonDeserialize(builder = HttpJsonOptions.HttpJsonOptionsBuilder.class)
+public class HttpJsonOptions {
+
+  @JsonInclude
+  private final boolean allowNanInf;
+
+  @JsonInclude
+  private final boolean allTextMode;
+
+  @JsonInclude
+  private final boolean readNumbersAsDouble;
+
+  @JsonInclude
+  private final boolean enableEscapeAnyChar;
+
+  /**
+   * Describes whether or not this reader can unwrap a single root array record
+   * and treat it like a set of distinct records.
+   */
+  @JsonInclude
+  private final boolean skipOuterList;  // Default should be true
+
+  @JsonIgnore
+  public JsonLoaderOptions getJsonOptions(OptionSet optionSet) {
+
+    JsonLoaderOptions options = new JsonLoaderOptions();
+
+    if (optionSet.getBoolean(ExecConstants.JSON_READER_NAN_INF_NUMBERS) != allowNanInf) {
+      options.allowNanInf = allowNanInf;
+    } else {
+      options.allowNanInf = optionSet.getBoolean(ExecConstants.JSON_READER_NAN_INF_NUMBERS);
+    }
+
+    if (optionSet.getBoolean(ExecConstants.JSON_ALL_TEXT_MODE) != allTextMode) {
+
+      options.allTextMode = allTextMode;
+    } else {
+      options.allTextMode = optionSet.getBoolean(ExecConstants.JSON_ALL_TEXT_MODE);
```

Comment:
It is fine to set `options.allTextMode = allTextMode`. But looks like the above `if-else` is redundant and can be simplified to `options.allTextMode = allTextMode;`.  Same for other options

Reply:
> It is fine to set `options.allTextMode = allTextMode`. But looks like the above `if-else` is redundant and can be simplified to `options.allTextMode = allTextMode;`. Same for other options

@vdiravka Thanks for the review.  I can remove the `else` block, since we are already importing the options, but what about the case where the user does not define `allTextMode` and the system setting is `true`?  In that case, if we don't have the if statement, the system setting will be unintentionally overwritten.

Reply:
@vdiravka Fixed.  Thanks!

Reply:
Just logically
```
    if (optionSet.getBoolean(ExecConstants.JSON_ALL_TEXT_MODE) != allTextMode) {
      options.allTextMode = allTextMode;
    } else {
      options.allTextMode = optionSet.getBoolean(ExecConstants.JSON_ALL_TEXT_MODE);
    }
```
fully equals to:
```
    options.allTextMode = allTextMode;
```

But you are right about missing option in plugin configs and non default system option. In this case we need take into account that system option. To do this we need understand, when the plugin config option is absent. It is not possible with primitive `boolean`, because `false` can mean missing plugin config value or user specified false value. So to fix this need to use Boolean. Therefore by default (when option is not specified in plugin config) it will be null and for this case we can use system/session option here.

Reply:
@vdiravka  Thanks for the feedback.  I changed the variable type to `Boolean` rather than the primitive.  But I still see three cases:
1.  User provided input does not match system settings.  In this case, we go with the user provided input.  
2. User provided input does match system settings.  In this case it doesn't really matter.
3. User did not provide input.  

To fix this, I refactored and used a `null` check for the input.

Reply:
Nice! I agree with the above cases